pkg/registry/admissionregistration/validatingwebhookconfiguration/strategy.go

```diff
@@ -63,7 +65,12 @@ func (validatingWebhookConfigurationStrategy) PrepareForUpdate(ctx context.Conte
 
 // Validate validates a new validatingWebhookConfiguration.
 func (validatingWebhookConfigurationStrategy) Validate(ctx context.Context, obj runtime.Object) field.ErrorList {
-	return validation.ValidateValidatingWebhookConfiguration(obj.(*admissionregistration.ValidatingWebhookConfiguration))
+	var groupVersion schema.GroupVersion
+	if requestInfo, found := genericapirequest.RequestInfoFrom(ctx); found {
+		groupVersion = schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}
```

Comment:
if we panic in the else case, what breaks? Only tests?

Reply:
I wanted to make that any internal code paths calling this would get the tight validation. Only the invocation that explicitly comes in via v1beta1 gets the looser validation. This is the same thing we do in other places:

https://github.com/kubernetes/kubernetes/blob/79aab0b90d9564dd3ae28b1764608c6dd0aa09f7/pkg/registry/storage/volumeattachment/strategy.go#L78-L90